### PR TITLE
Fixup documentation for consistency

### PR DESCRIPTION
**Pull request by schaefi**
There is the legacy kiwi version and there is this kiwi(next generation).
From a documentation perspective there are several inconsistencies that
could confuse users. This commit makes the name for KIWI-NG consistent
across the entire documentation. At places where we point to older
documentation we use the term Legacy KIWI and a link to the documentation
that covers this part. All this is needed in preparation to cleanup the
documentation situation for the SUSE documentation but with respect to
the upstream doc sources, their layout and markup.

### CODE REVIEW

File: doc/source/working_with_images/pxe_client_configuration.rst [hunk 1]
```diff
@@ -6,13 +6,13 @@ PXE Client Setup Configuration
...
    This page provides further information for handling
-   PXE images built with KIWI and references the following
+   PXE images built with {kiwi} and references the following
    articles:
 
    * :ref:`build_pxe`
 
 All PXE boot based deployment methods are controlled by configuration files
-located in :file:`/srv/tftpboot/KIWI` on the PXE server. Such a configuration
+located in :file:`/srv/tftpboot/{kiwi}` on the PXE server. Such a configuration
```
Comment (davidcassany) [inline doc/source/working_with_images/pxe_client_configuration.rst]: I doubt this is correct. This should not be translated to `KIWI-NG`, right?

File: doc/source/working_with_images/pxe_client_configuration.rst [hunk 2]
```diff
@@ -42,8 +42,8 @@ The configuration file has got the following general format:
 
     CONF="src;dest;srvip;bsize;[hash],...,src;dest;srvip;bsize;[hash]"
 
-    KIWI_BOOT_TIMEOUT="seconds"
-    KIWI_KERNEL_OPTIONS="opt1 opt2 ..."
+    {kiwi}_BOOT_TIMEOUT="seconds"
+    {kiwi}_KERNEL_OPTIONS="opt1 opt2 ..."
```
Comment (davidcassany) [inline doc/source/working_with_images/pxe_client_configuration.rst]: Same as above, those variables should not be translated to `KIWI-NG`

File: doc/source/working_with_images/pxe_client_configuration.rst [hunk 3]
```diff
@@ -207,7 +207,7 @@ boot of the client the following setup is required:
 
 .. code:: bash
 
-   KIWI_KERNEL_OPTIONS="nomodeset"
+   {kiwi}_KERNEL_OPTIONS="nomodeset"
```
Comment (davidcassany) [inline doc/source/working_with_images/pxe_client_configuration.rst]: Same as above

File: doc/source/working_with_images/pxe_client_configuration.rst [hunk 4]
```diff
@@ -223,7 +223,7 @@ the following setup is required.
 
 .. code:: bash
 
-   KIWI_BOOT_TIMEOUT="10"
+   {kiwi}_BOOT_TIMEOUT="10"
```
Comment (davidcassany) [inline doc/source/working_with_images/pxe_client_configuration.rst]: Same as above

File: doc/source/building/build_pxe_root_filesystem.rst [hunk 5]
```diff
@@ -118,7 +118,7 @@ system. As diskless client, a QEMU virtual machine is used.
 
    .. code:: bash
 
-       $ vi /srv/tftpboot/KIWI/config.default
+       $ vi /srv/tftpboot/{kiwi}/config.default
```
Comment (davidcassany) [inline doc/source/building/build_pxe_root_filesystem.rst]: I believe this should not be translated to `KIWI-NG`, just `kiwi`.

File: doc/source/building/build_pxe_root_filesystem.rst [hunk 6]
```diff
@@ -105,9 +105,9 @@ system. As diskless client, a QEMU virtual machine is used.
 
    .. code:: bash
 
-       DEFAULT KIWI-Boot
+       DEFAULT {kiwi}-Boot
```
Comment (davidcassany) [inline doc/source/building/build_pxe_root_filesystem.rst]: I guess the change doesn't really make a difference here. However I'd prefer to not use a variable here.

File: doc/source/working_with_kiwi/legacy_kiwi.rst [hunk 7]
```diff
@@ -39,25 +39,25 @@ In order to address all of these the questions came up:
...
 
-This provides an opportunity for users to test the next generation KIWI
+This provides an opportunity for users to test the next generation {kiwi}
 with their appliance descriptions without risk. If it builds and works
-as expected, I recommend to switch to the next generation KIWI. If not,
+as expected, I recommend to switch to the next generation {kiwi}. If not,
 please open an issue on https://github.com/OSInside/kiwi.
 
-The legacy KIWI version will be further developed in maintenance mode.
+The legacy {kiwi} version will be further developed in maintenance mode.
```
Comment (davidcassany) [inline doc/source/working_with_kiwi/legacy_kiwi.rst]: I guess this is inconsistent `legacy KIWI-NG` :wink:

File: doc/source/working_with_kiwi/legacy_kiwi.rst [hunk 8]
```diff
@@ -39,25 +39,25 @@ In order to address all of these the questions came up:
...
 There won't be any new features added in that code base though.
 Packages will be available at the known place:
-`Legacy KIWI packages <http://download.opensuse.org/repositories/Virtualization:/Appliances>`__
+`Legacy {kiwi} packages <http://download.opensuse.org/repositories/Virtualization:/Appliances>`__
```
Comment (davidcassany) [inline doc/source/working_with_kiwi/legacy_kiwi.rst]: Same as above, review the hole page, there are many `legacy {kiwi}` or `next generation {kiwi}` in this one. Probably we should rephrase some of those.

**Comment by schaefi** [comment]
Oh Oh yes you are right, I will fix all of those. Thanks much for the detailed check

File: doc/source/index.rst [hunk 9]
```diff
@@ -1,11 +1,11 @@
...
-==================
+{kiwi} Documentation
+=====================
 
 .. note::
 
-   This documentation covers KIWI |version|- the command line utility to
+   This documentation covers {kiwi} |version|- the command line utility to
    build Linux system appliances. For versions of KIWI older or equal to
    v7.x.x please refer to `Legacy KIWI <https://doc.opensuse.org/projects/kiwi/doc>`__
```
Comment (tomschr) [inline doc/source/index.rst]: I would rephrase the note to emphasize the difference between the two:


```rst
.. note:: KIWI-NG vs. Legacy KIWI

     This documentation covers {kiwi-full} ({kiwi}) |version| the command line
     utility to build Linux system appliances.
     {kiwi} is stable and all new ideas, bugfixes, and improvements
     will be developed here.
     Older {kiwi} versions older or equal to v7.x.x are out of
     maintenance and do not get any updates or bugfixes.
     If you still need this verson, refer to the documentation for
     Legacy KIWI.
```

File: doc/source/commands.rst [hunk 10]
```diff
@@ -1,9 +1,9 @@
-KIWI Commands
-=============
+Command Line
+============
 
 .. note::
 
-   This document provides a list of the existing KIWI commands
+   This document provides a list of the existing {kiwi} commands
```
Comment (tomschr) [inline doc/source/commands.rst]: As this will be a manpage at some point, I would add a short explanation about our acronym "NG", using the "full" name with the `{kiwi-full}` placeholder:

```suggestion
   This document provides a list of the existing {kiwi-full} ({kiwi}) commands
```

File: doc/source/conf.py [hunk 11]
```diff
@@ -74,11 +73,12 @@ def setup(app):
...
     '{exc_kiwi_repo}':
         'obs://Virtualization:Appliances:Builder/openSUSE_Leap_15.1',
-    '{schema_version}': '7.1'
+    '{schema_version}': '7.1',
+    '{kiwi}': 'KIWI-NG'
 }
 
 latex_documents = [
-    ('index', 'kiwi.tex', 'KIWI Documentation', 'Marcus Schäfer', 'manual')
+    ('index', 'kiwi.tex', 'KIWI-NG Documentation', 'Marcus Schäfer', 'manual')
```
Comment (tomschr) [inline doc/source/conf.py]:
```suggestion
    ('index', 'kiwi.tex', 'KIWI NG Documentation', 'Marcus Schäfer', 'manual')
```

File: doc/source/conf.py [hunk 12]
```diff
@@ -117,8 +117,8 @@ def setup(app):
 default_role="py:obj"
 
 # General information about the project.
-project = 'kiwi'
-copyright = '2019, Marcus Schäfer'
+project = 'KIWI-NG'
```
Comment (tomschr) [inline doc/source/conf.py]:
```suggestion
project = 'KIWI NG'
```

File: doc/source/conf.py [hunk 13]
```diff
@@ -74,11 +73,12 @@ def setup(app):
     '{exc_repo}': 'obs://openSUSE:Leap:15.1/standard',
     '{exc_kiwi_repo}':
         'obs://Virtualization:Appliances:Builder/openSUSE_Leap_15.1',
-    '{schema_version}': '7.1'
+    '{schema_version}': '7.1',
+    '{kiwi}': 'KIWI-NG'
```
Comment (tomschr) [inline doc/source/conf.py]: From our native speaker, she would replace the dash with a space. Additionally, I would also define the full name, without the "NG" acronym:

```suggestion
    '{kiwi}': 'KIWI NG',
    '{kiwi-full}': 'KIWI Next Generation',
```

File: doc/source/working_with_images/pxe_client_configuration.rst (hunk 1 above)
Comment (schaefi) [reply doc/source/working_with_images/pxe_client_configuration.rst]: fixed

File: doc/source/working_with_kiwi/legacy_kiwi.rst (hunk 7 above)
Comment (schaefi) [reply doc/source/working_with_kiwi/legacy_kiwi.rst]: yeah this is just stupid :-) fixed now

File: doc/source/commands.rst (hunk 10 above)
Comment (schaefi) [reply doc/source/commands.rst]: yes done and also updated the man pages. kiwi-ng is the actual name for the binary. Other names (kiw, kiwi-ng-3) are symlinks that can be used too. But the man pages should use the main name

File: doc/source/index.rst (hunk 9 above)
Comment (schaefi) [reply doc/source/index.rst]: I like it and used parts of it. Thanks

**Comment by schaefi** [comment]
@davidcassany @tomschr I think I have covered all issues, could you take a look again ? Thanks

File: doc/source/working_with_kiwi/legacy_kiwi.rst [hunk 14]
```diff
@@ -1,7 +1,7 @@
 .. _legacy_kiwi:
 
-Legacy KIWI vs. Next Generation
-===============================
+Legacy KIWI vs. {kiwi-product}
+==============================================
```
Comment (tomschr) [inline doc/source/working_with_kiwi/legacy_kiwi.rst]: You could remove some of the equal signs. :wink:

File: doc/source/working_with_kiwi/legacy_kiwi.rst [hunk 15]
```diff
@@ -39,25 +40,25 @@ In order to address all of these the questions came up:
...
+the same appliance description with the {kiwi-legacy} and the 
+{kiwi-product}, if the distribution and all configured features are
+supported by the used version.
 
-This provides an opportunity for users to test the next generation KIWI
+This provides an opportunity for users to test the {kiwi-product}
 with their appliance descriptions without risk. If it builds and works
-as expected, I recommend to switch to the next generation KIWI. If not,
+as expected, I recommend to switch to the {kiwi-product}. If not,
 please open an issue on https://github.com/OSInside/kiwi.
```
Comment (tomschr) [inline doc/source/working_with_kiwi/legacy_kiwi.rst]: In my opinion, `{kiwi-product}` should be used to *introduce* the term. After you have used it, you just use `{kiwi}`.

File: doc/source/working_with_kiwi/legacy_kiwi.rst (hunk 14 above)
Comment (schaefi) [reply doc/source/working_with_kiwi/legacy_kiwi.rst]: no you can't. {kiwi-product} is replaced with content prior the sphinx rendering. The content string is longer than the macro. If you delete the signs you break the rendering

File: doc/source/working_with_kiwi/legacy_kiwi.rst [hunk 16]
```diff
@@ -39,25 +40,25 @@ In order to address all of these the questions came up:
...
 
-The legacy KIWI version will be further developed in maintenance mode.
+The {kiwi-legacy} version will be further developed in maintenance mode.
 There won't be any new features added in that code base though.
 Packages will be available at the known place:
-`Legacy KIWI packages <http://download.opensuse.org/repositories/Virtualization:/Appliances>`__
+`{kiwi-legacy} packages <http://download.opensuse.org/repositories/Virtualization:/Appliances>`__
 
 When Do I need to use the legacy kiwi
```
Comment (tomschr) [inline doc/source/working_with_kiwi/legacy_kiwi.rst]:
```suggestion
When Do I need to use the {kiwi-legacy}
```

File: doc/source/working_with_kiwi/legacy_kiwi.rst (hunk 15 above)
Comment (schaefi) [reply doc/source/working_with_kiwi/legacy_kiwi.rst]: This applies to all other chapters but not the explanations we use in this legacy chapter. I suggest to build it and read the result. For me that looked ok

File: doc/source/working_with_kiwi/legacy_kiwi.rst [hunk 17]
```diff
@@ -141,15 +142,15 @@ generation KIWI.
...
-the next generation KIWI does not provide, the build will fail with an
+the {kiwi-product} does not provide, the build will fail with an
 exception early. If the appliance description uses next generation
 features like the selection of the initrd system, it's not possible to
-build that with the legacy KIWI, unless the appliance description
+build that with the {kiwi-legacy}, unless the appliance description
 properly encapsulates the differences into a profile.
 
-The next generation KIWI also provides the `--compat` option and
+The {kiwi-product} also provides the `--compat` option and
```
Comment (tomschr) [inline doc/source/working_with_kiwi/legacy_kiwi.rst]: Again, the product is already introduced so `{kiwi}` should be enough distinction against Legacy KIWI.

**Comment by tomschr** [comment]
@schaefi Looks great, I just found some minor details you can think about.

File: doc/source/working_with_kiwi/legacy_kiwi.rst (hunk 15 above)
Comment (schaefi) [reply doc/source/working_with_kiwi/legacy_kiwi.rst]: sorry I got it wrong, yes I agree will update the text

**Comment by schaefi** [comment]
@davidcassany @tomschr how about it now ? can we go from here

**Comment by tomschr** [comment]
> @tomschr how about it now ? can we go from here

I've built it and it looks good. There is one thing which was a bit irritating. Not sure if I got it right, but... the main script name has been changed to `kiwi-ng` and there is only a link to `kiwi` right?

So your intention is that the user calls the script by  `kiwi-ng`  (and not `kiwi`), correct? In that case you might want to adapt the help output.

I'm looking at the file `kiwi/cli.py` and the file docstring contains `kiwi` instead of `kiwi-ng`. Maybe you want to change that. :wink:

Other than that, I haven't found anything else. So go for it! :rocket:

**Comment by schaefi** [comment]
>I'm looking at the file kiwi/cli.py and the file docstring contains kiwi instead of kiwi-ng. Maybe you want to change that. ￼

That's hard to change because the docstring is used with docopt and the api structure is kiwi/... not kiwi-ng. The name kiwi-ng comes from setup.py

```python
'entry_points': {
        'console_scripts': [
            'kiwi-ng=kiwi.kiwi:main',
            'kiwicompat=kiwi.kiwi_compat:main'
        ]
    }
```

So don't get me wrong but I rather risk this type of inconsistency before I break the entire cli handling. Hope that makes sense to you too ?

**Comment by schaefi** [comment]
There are also more docopt strings. See the code in kiwi/tasks/*

**Comment by schaefi** [comment]
@davidcassany What's is your opinion on that ? The man pages with this changes here will document the kiwi call as "kiwi-ng ..." and therefore matches the entry point from setup.py. The docstrings relevant for docopt needs to follow the code structure iirc which is kiwi/... Thus a kiwi-ng --help or a kiwi-ng-3 --help or a kiwi --help will only talk about "kiwi"

I thought too long about this... so happy if someone else has some input

**Comment by schaefi** [comment]
maybe we just revert the kiwi-ng change in the man pages...

**Comment by tomschr** [comment]
> So don't get me wrong but I rather risk this type of inconsistency before I break the entire cli handling. Hope that makes sense to you too ?

Ahh right! Thanks for the explanation! :+1: I didn't realize that. If that is too difficult to change, then just leave it as is for the time being. We can change that (if really wanted) in a different pull request.
